File: BAEKJOON/basic_algo_lecture/1874.swift
let N = Int(readLine()!)!

var count = 1
var stack: [Int] = []
var answer: [String] = []
var hasAnswer = true

for _ in 0..<N {
	let n = Int(readLine()!)!
	while count <= n {
		stack.append(count)
		answer.append("+")
		count += 1
	}
	if n == stack.last! {
		stack.removeLast()
		answer.append("-")
	}
	else {
		hasAnswer = false
	}
}

if hasAnswer {
	print(answer.joined(separator: "\n"))
}
else {
	print("NO")
}

File: BAEKJOON/basic_algo_lecture/0x05_stack/1874.js
const fs = require("fs");
const input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((ele) => Number(ele));
const n = input[0];

let count = 1;
let stack = [];
let answer = [];
let hasAnswer = true;

for (let i = 0; i < n; i++) {
  const number = input[i + 1];
  while (number >= count) {
    stack.push(count);
    answer.push("+");
    count += 1;
  }
  if (number == stack[stack.length - 1]) {
    stack.pop();
    answer.push("-");
  } else {
    hasAnswer = false;
  }
}

if (hasAnswer) console.log(answer.join("\n"));
else console.log("NO");
